### PR TITLE
New matrix library

### PR DESCRIPTION
**Pull request by blerner**

### CODE REVIEW

File: tests/pyret/tests/test-matrices2.arr [hunk 1]
```diff
@@ -0,0 +1,862 @@
+include matrices
```
Comment (jpolitz) [inline tests/pyret/tests/test-matrices2.arr:1]: Should we add a comment or comments that link back to Racket sources?

File: tests/pyret/tests/test-equality.arr [hunk 2]
```diff
@@ -24,13 +24,23 @@ check "numbers":
   roughly-equal(3, 3) is true
   roughly-equal(3, 3.000003) is true
   roughly-equal-now(3, 3.000003) is true
-  roughly-equal-now(3, 3.000004) is false
```
Comment (jpolitz) [inline tests/pyret/tests/test-equality.arr:27]: Should this be factored out into a separate PR?

**Comment by blerner** [comment]
Note: need to update checker library to use within instead of within-rel, to match intended new behavior

File: tests/pyret/tests/test-equality.arr (hunk 2 above)
Comment (blerner) [reply tests/pyret/tests/test-equality.arr:27]: _maaaaaaaaaaaaaybe_, but I included it here because getting any sort of roughly-equals behavior on matrices _really needs_ a better notion of "roughly equal to zero": you simply can't say something is "roughly the identity matrix" without having a more tolerant notion of "roughly equal to zero".
